### PR TITLE
machines: Fix OS autodetection in create VM dialog and add test for it

### PR DESCRIPTION
**Pull request by KKoukiou**
Related to https://bugzilla.redhat.com/show_bug.cgi?id=1862333

### CODE REVIEW

File: test/verify/check-machines [hunk 1]
```diff
@@ -1390,6 +1390,16 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
                                                       os_name=config.FEDORA_28,
                                                       start_vm=False))
 
+        virtInstallVersion = self.machine.execute("virt-install --version")
+        if virtInstallVersion >= "2":
```
Comment (martinpitt) [inline test/verify/check-machines:1394]: I'm not a big fan of such dynamic checks. They get silently skipped if the condition is somehow wrong, or the output format changes, etc. It'd be better to have an explicit list of OSes where it does *not* work, and also cover the "else:" behaviour.

Comment (KKoukiou) [reply test/verify/check-machines]: Fixed. Also fixed in another place which followed this approach.

File: test/verify/check-machines [hunk 2]
```diff
@@ -2019,7 +2015,13 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
 
             # For the mocked fedora URL installation wait to see that the OS section is automatically filed
             if self.sourceType == "url" and self.location == TestMachines.TestCreateConfig.TREE_URL:
-                b.wait_attr("label[for=os-select] + div input", "value", TestMachines.TestCreateConfig.FEDORA_28)
+                # virtInstallVersion >= "2" is needed for OS autodetection
```
Comment (martinpitt) [inline test/verify/check-machines:2018]: But then this comment doesn't make sense -- all Debian/Ubuntu versions have virt-manager > 2 (see above), so the reason for teh ubuntu/debian special-casing must be something else.

File: test/verify/check-machines [hunk 3]
```diff
@@ -1390,15 +1390,13 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
                                                       os_name=config.FEDORA_28,
                                                       start_vm=False))
 
-        virtInstallVersion = self.machine.execute("virt-install --version")
-        if virtInstallVersion >= "2":
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply test/verify/check-machines]: Oh it's not about virt-install  of course - I overlooked the comment.
os autodetection is only about libosinfo which for the above two OSes is not sufficient version to support this feature.

**Pull request merged**